crates/config/src/providers/ext.rs

```diff
@@ -80,22 +80,118 @@ impl TomlFileProvider {
     }
 
     fn read(&self) -> Result<Map<Profile, Dict>, Error> {
-        use serde::de::Error as _;
-        if let Some(file) = self.env_val() {
-            let path = Path::new(&file);
-            if !path.exists() {
+        use serde::{Deserialize, de::Error as _};
+        use std::collections::HashMap;
+
+        #[derive(Deserialize, Default)]
+        struct ExtendConfig {
+            #[serde(default)]
+            extends: Option<PathBuf>,
+        }
+
+        #[derive(Deserialize, Default)]
+        struct PartialConfig {
+            #[serde(default)]
+            profile: Option<HashMap<String, ExtendConfig>>,
+        }
+
+        let local_path = self.file();
+        if !local_path.exists() {
+            if let Some(file) = self.env_val() {
                 return Err(Error::custom(format!(
                     "Config file `{}` set in env var `{}` does not exist",
                     file,
                     self.env_var.unwrap()
                 )));
             }
-            Toml::file(file)
+            return Ok(Map::new());
+        }
+
+        let local_provider = Toml::file(local_path.clone()).nested();
+
+        let local_path_str = local_path.to_string_lossy();
+        let local_content = std::fs::read_to_string(&local_path)
+            .map_err(|e| Error::custom(e.to_string()).with_path(&local_path_str))?;
+        let partial_config: PartialConfig = toml::from_str(&local_content)
+            .map_err(|e| Error::custom(e.to_string()).with_path(&local_path_str))?;
+
+        // Get the currently selected profile
+        let selected_profile = Config::selected_profile();
+
+        // Check if the current profile has an inherit_from field
+        let inherit_from = partial_config
```

Comment:
can we document what this new section does exactly, otherwise we'll eventually have a hard time figuring out what this does in the future